cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1162,17 +1203,65 @@ private boolean isPodToRestart(StatefulSet ss, Pod pod, boolean isAncillaryCmCha
                     reasons.add("Pod has old generation");
                 }
                 if (!reasons.isEmpty()) {
-                    log.debug("{}: Rolling pod {} due to {}",
-                            reconciliation, pod.getMetadata().getName(), reasons);
+                    if (isSatisfiedBy) {
+                        log.debug("{}: Rolling pod {} due to {}",
+                                reconciliation, pod.getMetadata().getName(), reasons);
+                    } else {
+                        log.debug("{}: Potential pod {} rolling due to {} but maintenance time windows not satisfied",
+                                reconciliation, pod.getMetadata().getName(), reasons);
+                    }
                 }
             }
-            return !isPodUpToDate || !isPodCaCertUpToDate || isAncillaryCmChange || isCaCertsChanged;
+            return isSatisfiedBy && isPodToRestart;
+        }
+
+        private boolean isMaintenanceTimeWindowsSatisfied() {
+            try {
+                boolean isSatisfiedBy = getMaintenanceTimeWindows() == null || getMaintenanceTimeWindows().isEmpty();
+                if (!isSatisfiedBy) {
+                    Date date = new Date();
+                    for (String cron : getMaintenanceTimeWindows()) {
+                        CronExpression cronExpression = new CronExpression(cron);
+                        // the user defines the cron expression in "his" timezone but CO pod
+                        // can be running on a different one, so setting it on the cron expression
+                        cronExpression.setTimeZone(TimeZone.getDefault());
+                        if (cronExpression.isSatisfiedBy(date)) {
+                            isSatisfiedBy = true;
+                            break;
+                        }
+                    }
+                }
+                return isSatisfiedBy;
+            } catch (ParseException e) {
+                log.warn("The provided maintenance time windows list contains a not valid cron expression");
```

Comment:
We should include the invalid expression(s). It would be annoying to have, say 20 expressions and not be told which was the bad one.